components/filter-dropdown/filter-dropdown.js

```diff
@@ -181,6 +218,38 @@ class D2LLabsFilterDropdown extends mixinBehaviors([LocalizeBehavior], PolymerEl
 		return this._localizeOrAlt(openerTextMultiple, 'filterMultiple', 'numOptions', totalSelectedOptionCount);
 	}
 
+	async _handleCategoriesSlotChange() {
+		if (!this._useNewStructure) return;
+
+		const tabsContainer = this.shadowRoot.querySelector('d2l-tabs');
+		const categories = this.querySelectorAll('d2l-labs-filter-dropdown-category');
+
+		if (!tabsContainer || !categories || categories.length === 0) return;
+
+		// Build tabInfos which is used to render d2l-tab components
+		const tabInfos = [];
+		categories.forEach((category, index) => {
+			const key = category.getAttribute('key') || `category-${index}`;
+			const text = category.getAttribute('text');
+
+			category.setAttribute('labelled-by', key);
+			tabInfos.push({ key, text });
+
+			// Observe updates to text
+			if (!category.__tabInfoObserver) {
+				category.__tabInfoObserver = new MutationObserver(() => {
+					const text2 = category.getAttribute('text');
+					this._tabInfos = this._tabInfos.map(info => {
+						return info.key === key ? { ...info, text: text2 } : info;
+					});
+				});
+				category.__tabInfoObserver.observe(category, { attributes: true, attributeFilter: ['text'] });
```

Comment:
TODO: `selected` should also be observed in case it's being set on the filter category.